presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaPageSourceProvider.java

```diff
@@ -254,17 +256,19 @@ private InternalTable buildPartitions(Session session, String catalogName, Map<S
                         String columnName = columnHandles.get(columnHandle);
                         String value = null;
                         if (entry.getValue().getValue() != null) {
-                            ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle.get(), columnHandle);
-                            try {
-                                Signature operator = metadata.getFunctionRegistry().getCoercion(columnMetadata.getType(), VARCHAR);
-                                value = ((Slice) metadata.getFunctionRegistry().getScalarFunctionImplementation(operator).getMethodHandle().invokeWithArguments(entry.getValue().getValue())).toStringUtf8();
-                            }
-                            catch (OperatorNotFoundException e) {
-                                value = "<UNREPRESENTABLE VALUE>";
-                            }
-                            catch (Throwable throwable) {
-                                throw Throwables.propagate(throwable);
-                            }
+                            value = values.computeIfAbsent(columnHandle, column -> {
```

Comment:
Actually, since you already have the `columnHandles` map above, make it a `Map<ColumnHandle, MethodHandle>` and pre-build the map rather than doing it on-demand. Then you can move `ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle.get(), column);` into that pre-build step too